chain-abci/src/staking/table.rs

```diff
@@ -168,11 +168,13 @@ impl StakingTable {
             // liveness and heap and idx_* are always consistent
             let mut staking = heap.get(addr).unwrap();
             assert!(self.idx_sort.insert((&staking).into()));
-            let val = staking.validator.as_mut().unwrap();
-            assert!(self
-                .idx_validator_address
-                .insert(val.validator_address(), *addr)
-                .is_none());
+            if let Some(NodeState::CouncilNode(val)) = staking.node_meta.as_mut() {
+                assert!(self
+                    .idx_validator_address
+                    .insert(val.validator_address(), *addr)
+                    .is_none());
+            }
+            // FIXME: else panic? unreachable?
```

Comment:
I think the invariant now is like: the `idx_*` only contains `CouncilNode` not `CommunityNode`, so it should panic here?

Reply:
ok, added a top level comment for this new invariant